packages/console-types/src/types/services.ts

```diff
@@ -132,21 +133,79 @@ const configMapEnv = {
 } as const
 export type EnvironmentVariablesFromConfigMap = FromSchema<typeof configMapEnv>
 
+const DOWNWARD_API_FIELDS = {
+  POD_LABELS: {
+    type: 'string',
+    pattern: "^metadata.labels\\['[a-zA-Z0-9-_.]+'\\]$",
+  },
+  POD_ANNOTATIONS: {
+    type: 'string',
+    pattern: "^metadata.annotations\\['[a-zA-Z0-9-_.]+'\\]$",
+  },
+  POD: [
```

Comment:
wouldn't an enum be more helpful?

Reply:
At that point we just need an array of string literals to put in the schema, how would you use an enum?

Reply:
I added enums for `POD` and `CONTAINER`, named respectively `DownwardAPIPodPath` and `DownwardAPIContainerPath`. To the `DownwardAPIPodPath` enum I also added two enums with values `metadata.annotations` and `metadata.labels`, without including them in the json schema enum, since they are part of a string defined with a regex in the schema. 
Putting them in the exported enum makes them available to library users to use them as string prefixes at runtime.

What do you think about this solution?